### PR TITLE
Update Azure Virtual desktop Scaling Plan Cmdlets to version 2021-09-03-preview

### PR DESCRIPTION
**Pull request by dallinlmag**
## Description

We needed to change the format of the schedule object attached to scaling plans in Desktop Virtualization.
We are in public preview, and the changes have been added to the rest api. The powershell changes were done using autorest. We discussed with the powershell team about this breaking change and were given permission to continue because we don't have any customers using the powershell at the moment.

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [ ] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by dallinlmag** [comment]
This is related to this PR https://github.com/Azure/azure-powershell/pull/16584.

**Comment by msJinLei** [comment]
```
  D:\a\1\s\artifacts//StaticAnalysisResults\BreakingChangeIssues.csv Errors
  "AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
  "Az.DesktopVirtualization","Get-AzWvdDesktop","Get-AzWvdDesktop","0","1020","The cmdlet 'Get-AzWvdDesktop' no longer has output type 'Microsoft.Azure.PowerShell.Cmdlets.DesktopVirtualization.Models.Api20210712.IDesktopList'.","Make cmdlet 'Get-AzWvdDesktop' return type 'Microsoft.Azure.PowerShell.Cmdlets.DesktopVirtualization.Models.Api20210712.IDesktopList'."
  "Az.DesktopVirtualization","Get-AzWvdRegistrationInfo","Get-AzWvdRegistrationInfo","0","1020","The cmdlet 'Get-AzWvdRegistrationInfo' no longer has output type 'Microsoft.Azure.PowerShell.Cmdlets.DesktopVirtualization.Models.Api20210712.RegistrationInfo'.","Make cmdlet 'Get-AzWvdRegistrationInfo' return type 'Microsoft.Azure.PowerShell.Cmdlets.DesktopVirtualization.Models.Api20210712.RegistrationInfo'."
  "Az.DesktopVirtualization","New-AzWvdScalingPlan","New-AzWvdScalingPlan","0","2020","The cmdlet 'New-AzWvdScalingPlan' no longer supports the type 'Microsoft.Azure.PowerShell.Cmdlets.DesktopVirtualization.Support.HostPoolType' for parameter 'HostPoolType'.","Change the type for parameter 'HostPoolType' back to 'Microsoft.Azure.PowerShell.Cmdlets.DesktopVirtualization.Support.HostPoolType'."
  "Az.DesktopVirtualization","Update-AzWvdScalingPlan","Update-AzWvdScalingPlan","0","2000","The cmdlet 'Update-AzWvdScalingPlan' no longer supports the parameter 'HostPoolType' and no alias was found for the original parameter name.","Add the parameter 'HostPoolType' back to the cmdlet 'Update-AzWvdScalingPlan', or add an alias to the original parameter name."
  "Az.DesktopVirtualization","Update-AzWvdScalingPlan","Update-AzWvdScalingPlan","0","1050","The parameter set 'UpdateExpanded' for cmdlet 'Update-AzWvdScalingPlan' has been removed.","Add parameter set 'UpdateExpanded' back to cmdlet 'Update-AzWvdScalingPlan'."
  "Az.DesktopVirtualization","Update-AzWvdScalingPlan","Update-AzWvdScalingPlan","0","1050","The parameter set 'UpdateViaIdentityExpanded' for cmdlet 'Update-AzWvdScalingPlan' has been removed.","Add parameter set 'UpdateViaIdentityExpanded' back to cmdlet 'Update-AzWvdScalingPlan'."
  "Az.DesktopVirtualization","Update-AzWvdScalingPlan","Update-AzWvdScalingPlan","0","1050","The parameter set '__AllParameterSets' for cmdlet 'Update-AzWvdScalingPlan' has been removed.","Add parameter set '__AllParameterSets' back to cmdlet 'Update-AzWvdScalingPlan'."
  
  D:\a\1\s\artifacts//StaticAnalysisResults\SignatureIssues.csv Errors
  "AssemblyFileName","ClassName","Target","Severity","ProblemId","Description","Remediation"
  "Az.DesktopVirtualization","Get-AzWvdRegistrationInfo","Get-AzWvdRegistrationInfo","1","8600","Cmdlet 'Get-AzWvdRegistrationInfo' has no defined output type.","Add an OutputType attribute that declares the type of the object(s) returned by this cmdlet. If this cmdlet returns no output, please set the output type to 'bool' and make sure to implement the 'PassThru' parameter."
  ```

**Comment by msJinLei** [comment]
@dallinlmag please submit a cmdlet review first https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues

**Comment by dallinlmag** [comment]
@msJinLei  cmdlet review submitted here: https://github.com/Azure/azure-powershell-cmdlet-review-pr/issues/1139 thanks

**Comment by BethanyZhou** [comment]
CI failed. Please have a look.

File: src/DesktopVirtualization/docs/Get-AzWvdPrivateEndpointConnection.md [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+---
+
+# Get-AzWvdPrivateEndpointConnection
+
+## SYNOPSIS
+Get a private endpoint connection.
+
+## SYNTAX
+
+### List (Default)
```
[the remark this answers is not in the document]
Comment (dallinlmag) [reply src/DesktopVirtualization/docs/Get-AzWvdPrivateEndpointConnection.md:15]: I have added the configuration, but we are currently branching from generation, so I have added it in another PR: https://github.com/Azure/azure-powershell/pull/17311

File: src/DesktopVirtualization/examples/Get-AzWvdPrivateEndpointConnection.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
+### Example 1: {{ Add title here }}
+```powershell
+PS C:\> {{ Add code here }}
```
Comment (BethanyZhou) [inline src/DesktopVirtualization/examples/Get-AzWvdPrivateEndpointConnection.md:3]: Please remove these files for absent commands

Comment (dallinlmag) [reply src/DesktopVirtualization/examples/Get-AzWvdPrivateEndpointConnection.md]: I removed all files for Get-AzWvdPrivateLinkResource, Get-AzWvdPrivateEndpointConnection, and Remove-AzWvdPrivateEndpointConnection.
 
Would you like me to remove all references from the docs? Or are they still accessible, just managed by the network team?

**Comment by BethanyZhou** [comment]
Hi @dallinlmag , please remove or hide these cmdlets by directives in README.md. If not, these files will still be generated.

File: src/DesktopVirtualization/examples/Get-AzWvdPrivateEndpointConnection.md (hunk 2 above)
Comment (dallinlmag) [reply src/DesktopVirtualization/examples/Get-AzWvdPrivateEndpointConnection.md]: @BethanyZhou Thanks for the idea. I added directives to hide the commands, and it no longer generates the files.

**Comment by dallinlmag** [comment]
@BethanyZhou I have removed the files that you asked to delete. Is there anything else that needs to be addressed before this is accepted?

**Comment by BethanyZhou** [comment]
Could you provide a change log for your change?

**Comment by isra-fel** [comment]
Hi @dallinlmag , 3 things left before we can merge this:
- Please check all the scripts in `custom/` and make sure all the api versions are up to date. [Here's an example where it's not.](https://github.com/Azure/azure-powershell/blob/29ed662f2b024c1faf7033e649a12c58d87c3592/src/DesktopVirtualization/custom/Get-AzWvdRegistrationInfo.ps1#L23)
- Please follow [these instructions](https://github.com/Azure/azure-powershell/blob/main/documentation/Debugging-StaticAnalysis-Errors.md) to suppress the error messages about breaking changes
- We update the changelog whenever there's a new release, so please write something in the comments which we can put in https://github.com/Azure/azure-powershell/blob/main/src/DesktopVirtualization/ChangeLog.md

**Comment by isra-fel** [comment]
Kindly ping @dallinlmag

**Comment by isra-fel** [comment]
Closing the PR for lack of activity. Feel free to reopen to continue working on it. Thanks 😀

**Pull request closed without merging**